### ISSUE TITLE
Zoom Last and Zoom Next Functions Should ignore Zoom Extents that are Overlapping

### ISSUE THREAD

**Issue by ar-siddiqui** [qgis/QGIS#38162]
**Feature description.**
QGIS would rock more if the Zoom Last and Zoom Next Functions ignore the Zoom Extents that are Overlapping. 

**Additional context**
The Zoom Last and Next Functions are not much useful in their current form. Usually, when you are working in QGIS, you pan multiple times as you work on the Canvas. The zoom last/next function remembers about 20 last pan/zoom extents. Which are mostly within the same or nearby locations. Users most of the time don't want to pan to same areas at different zoom levels but to completely different areas, but the zoom last/next function most of the time don't remember those extents because the allotted 20 extents are all within/nearby the same location you are working.

I would suggest that zoom last/next function ignore those extents that are overlapping.

See example:

https://drive.google.com/file/d/1ftRzWJeLMcVsEfbzodbyrwQ9IBDdsHiL/view?usp=sharing

**Comment by uclaros** [comment]
Zoom last/next are super useful as they are, especially if combined with the *zoom window* tool and bound to the mouse back/forward buttons!
By your description it seems you could better use spatial bookmarks (which are also super useful!)

**Comment by ar-siddiqui** [comment]
Spatial Bookmarks are for a completely different purpose. Zoom last/next are useful as they are right now, but they should either remember more than 20 previous extents (100 I would say at least) or they should ignore the almost similar extents that you get because of constant panning and zooming in and out. 20 previous extents are nothing when you are constantly moving in the map. I hope I am able to explain it better this time.

**Comment by uclaros** [comment]
I agree that the amount of extents remembered could be raised. However, I think you should practice more effective panning. There is no reason to do so many tiny pans within almost the same extents!
